<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="/home/khaled/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Turtlebot3 -->
  <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_world.launch" />
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
      <arg name="model" value="$(arg model)" />
  </include>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- move_base -->
  <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
      <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>

  <!-- Run the Perfect Localiser -->
  <node pkg="final_year_project" type="perfect_localiser.py" name="perfect_localiser"/>

  <!-- mcl -->
  <node pkg="monte_carlo_localisation" type="monte_carlo_localisation_node" name="monte_carlo_localisation_node" output="screen">
    <param name="init_x" value="-2.0"/>
    <param name="init_y" value="-0.5"/>
    <param name="alpha1" value="0.1"/>
    <param name="alpha2" value="0.1"/>
    <param name="alpha3" value="0.1"/>
    <param name="alpha4" value="0.1"/>
    <param name="z_hit" value="0.5"/>
    <param name="z_rand" value="0.5"/>
    <param name="kld_error" value="0.02"/>
    <param name="max_number_of_beams" value="180"/>
    <param name="odom_frame_id" value="odom"/>
    <param name="base_frame_id" value="base_link"/>
    <param name="laser_topic" value="scan"/>
  </node>

</launch>
